fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/codec/LuceneOptimized60FieldInfosFormat.java

```diff
@@ -0,0 +1,328 @@
+ *       <li>If the third lowest-order bit is set (0x4), payloads are stored for the
+ *           indexed field.</li>
+ *     </ul>
+ *   </li>
+ *   <li>IndexOptions: a byte containing index options.
+ *     <ul>
+ *       <li>0: not indexed</li>
+ *       <li>1: indexed as DOCS_ONLY</li>
+ *       <li>2: indexed as DOCS_AND_FREQS</li>
+ *       <li>3: indexed as DOCS_AND_FREQS_AND_POSITIONS</li>
+ *       <li>4: indexed as DOCS_AND_FREQS_AND_POSITIONS_AND_OFFSETS</li>
+ *     </ul>
+ *   </li>
+ *   <li>DocValuesBits: a byte containing per-document value types. The type
+ *       recorded as two four-bit integers, with the high-order bits representing
+ *       <code>norms</code> options, and the low-order bits representing
+ *       {@code DocValues} options. Each four-bit integer can be decoded as such:
+ *     <ul>
+ *       <li>0: no DocValues for this field.</li>
+ *       <li>1: NumericDocValues. ({@link DocValuesType#NUMERIC})</li>
+ *       <li>2: BinaryDocValues. ({@code DocValuesType#BINARY})</li>
+ *       <li>3: SortedDocValues. ({@code DocValuesType#SORTED})</li>
+ *      </ul>
+ *   </li>
+ *   <li>DocValuesGen is the generation count of the field's DocValues. If this is -1,
+ *       there are no DocValues updates to that field. Anything above zero means there
+ *       are updates stored by {@link DocValuesFormat}.</li>
+ *   <li>Attributes: a key-value map of codec-private attributes.</li>
+ *   <li>PointDimensionCount, PointNumBytes: these are non-zero only if the field is
+ *       indexed as points, e.g. using {@link org.apache.lucene.document.LongPoint}</li>
+ * </ul>
+ *
+ */
+public final class LuceneOptimized60FieldInfosFormat extends FieldInfosFormat {
+    /** Extension of field infos. */
+    static final String EXTENSION = "fnm";
+
+    // Codec header
+    static final String CODEC_NAME = "Lucene60FieldInfos";
+    static final int FORMAT_START = 0;
+    static final int FORMAT_SOFT_DELETES = 1;
+    static final int FORMAT_SELECTIVE_INDEXING = 2;
+    static final int FORMAT_CURRENT = FORMAT_SELECTIVE_INDEXING;
+
+    // Field flags
+    static final byte STORE_TERMVECTOR = 0x1;
+    static final byte OMIT_NORMS = 0x2;
+    static final byte STORE_PAYLOADS = 0x4;
+    static final byte SOFT_DELETES_FIELD = 0x8;
+
+    /** Sole constructor. */
+    public LuceneOptimized60FieldInfosFormat() {
+    }
+
+    @Override
+    public FieldInfos read(Directory directory, SegmentInfo segmentInfo, String segmentSuffix, IOContext context) throws IOException {
+        final String fileName = IndexFileNames.segmentFileName(segmentInfo.name, segmentSuffix, EXTENSION);
+        try (ChecksumIndexInput input = directory.openChecksumInput(fileName, context)) {
+            Throwable priorE = null;
+            FieldInfo[] infos = null;
+            try {
+                int version = CodecUtil.checkHeader(input,
+                        CODEC_NAME,
+                        FORMAT_START,
+                        FORMAT_CURRENT);
```

Comment:
There needs to be some way of conveying that -- other than standardizing formatting -- this and the corresponding `write` header change are all that is different from the original `Lucene60FieldInfosFormat`.